src/qmapshack/gis/prj/IGisProject.cpp

```diff
@@ -1215,3 +1245,26 @@ void IGisProject::gainUserFocus(bool yes)
         keyUserFocus.clear();
     }
 }
+
+CProjectFilterItem* IGisProject::filterProject(bool filter)
+{
+    if(filter)
+    {
+        if(projectFilter == nullptr)
+        {
+            projectFilter = new CProjectFilterItem(this);
+            insertChild(0, projectFilter);
+        }
+        //Set expanded anyways to show that search exists
+        projectFilter->showLineEdit();
+        setExpanded(true);
+    }
+    else
+    {
+        removeChild(projectFilter);
+        delete projectFilter;
```

Comment:
As far as I can see, this is not possible since CProjectFilterItem is not derived from QObject and according to qt's documentation you can use delete to a QTreeWidgetItem anytime.

Reply:
right, it's a `QTreeWidgetItem`. 

But the root problem is still existing: `CSearchLineEdit::slotClearFilter()` calls `connectedProject->filterProject(false);`  The will delete the `QTreeWidgetItem` that owns the `CSearchLineEdit`. From my understanding this will delete the `CSearchLineEdit` wile it is still in `slotClearFilter`. It depends a bit how this is coded in Qt, but nevertheless we shouldn't expect or count on anything. This is risky code. On Linux such stuff usually works, on Windows it often fails.

And from my point of view it's not necessary. Pressing the `X` in the filter shouldn't remove it. It just should clear it.